app/main.py

```diff
@@ -1,3 +1,54 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, km: float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Distance) -> Distance:
+        km = other.km if isinstance(other, Distance) else other
+        return Distance(
+            km=self.km + km
+        )
+
+    def __iadd__(self, other: Distance) -> Distance:
```

Comment:
Similar to the `__add__` method, the type hint for the `other` parameter in the `__iadd__` method should allow for both `Distance` and `float` types. Consider using `Union[Distance, float]`.